src/axom/primal/operators/detail/in_curved_polygon_impl.hpp

```diff
@@ -25,82 +25,119 @@ namespace primal
  * \param [in] c The BezierCurve object to compute the winding number along
+ * \param [in] edge_tol The physical distance level which objects are indistinguishable
+ * \param [in] EPS Miscellaneous numerical tolerance for isNearlyZero
+ * \pre Control polygon for c must be convex
  *
  * The winding number for a Bezier curve with a convex control polygon is
  * given by the signed angle between the tangent vector at that endpoint and
- * the vector in the direction of the other endpoint. The query can be located
- * on both endpoints if it is closed.
+ * the vector in the direction of the other endpoint. 
+ * 
+ * The query can be located on both endpoints if it is closed, in which case
+ * the angle is that between the tangent lines at both endpoints
  * 
  * \return 
  */
 template <typename T>
-double convex_endpoint_winding_number(bool is_init, const BezierCurve<T, 2>& c)
+double convex_endpoint_winding_number(const Point<T, 2>& q,
+                                      const BezierCurve<T, 2>& c,
+                                      double edge_tol,
+                                      double EPS)
 {
   const int ord = c.getOrder();
 
   if(ord == 1) return 0;
 
-  Vector<T, 2> V1, V2;
-  if(is_init)  // Query point is on initial endpoint
-  {
-    V1 = Vector<T, 2>(c[0], c[1]).unitVector();
-    V2 = Vector<T, 2>(c[0], c[ord]).unitVector();
-  }
-  else  // Query point is on terminal endpoint
-  {
-    V1 = Vector<T, 2>(c[ord], c[0]).unitVector();
-    V2 = Vector<T, 2>(c[ord], c[ord - 1]).unitVector();
-  }
+  int idx;
+
+  // Need to find vectors that subtend the entire curve.
+  //   We must ignore duplicate nodes
+  for(idx = 0; idx <= ord; ++idx)
+    if(squared_distance(q, c[idx]) > edge_tol * edge_tol) break;
+  Vector<T, 2> V1(q, c[idx]);
+
+  for(idx = ord; idx >= 0; --idx)
+    if(squared_distance(q, c[idx]) > edge_tol * edge_tol) break;
+  Vector<T, 2> V2(q, c[idx]);
 
   // clang-format off
   double orient = axom::numerics::determinant(V1[0] - V2[0], V2[0], 
                                               V1[1] - V2[1], V2[1]);
   // clang-format on
 
-  // If the tangent lines are parallel, winding number is ambiguous
-  if(axom::utilities::isNearlyEqual(orient, 0.0)) return 0;
-
-  double dotprod = axom::utilities::clampVal(V1.dot(V2), -1.0, 1.0);
+  // This means the bounding vectors are anti-parallel.
+  //  Parallel tangents can't happen with nontrivial convex control polygons
+  if(axom::utilities::isNearlyEqual(orient, 0.0, EPS))
+  {
+    for(int i = 1; i < ord; ++i)
+    {
+      // Need to find the first non-parallel control node
+      V2 = Vector<T, 2>(q, c[i]);
+
+      // clang-format off
+      double orient = axom::numerics::determinant(V1[0] - V2[0], V2[0], 
+                                                  V1[1] - V2[1], V2[1]);
+      // clang-format on
+
+      // Because we are convex, a single non-colinear vertex tells us the orientation
```

Comment:
```suggestion
      // Because we are convex, a single non-collinear vertex tells us the orientation
```

Reply:
Looking this up in a couple of online dictionaries, I think either spelling is acceptable?

Reply:
Whoops, I've been trying to stick to collinear. Thanks for catching it!